relm4/src/sender.rs

```diff
@@ -124,7 +149,22 @@ impl<C: FactoryComponent> FactoryComponentSender<C> {
         self.shared.command(cmd);
     }
 
-    /// Spawns a future that will be dropped as soon as the component is shut down.
+    /// Spawns a synchronous command.
+    ///
+    /// This is particularly useful for CPU-intensive background jobs that
+    /// need to run on a thread-pool in the background.
+    ///
+    /// The command will always complete, so it's better to
+    /// call [`try_send`](Sender::try_send) if you expect the component
+    /// to be dropped while the command is running.
```

Comment:
This seems like a potential footgun. Won't this always be a problem because the user could close the application at any time?

Reply:
You're right, this isn't a great solutions.

A solution to this problem could be the following: Offer another sender type (maybe called `TrySender`) that is used only for `CommandOutput` whereas `Input` and `Output` still use the current sender implementation. The idea behind this is that `Input` and `Output` are hardly ever used after a component was destroyed (even if it is possible in theory). Sending a `CommandOutput` on the other side should probably return a `Result` by default so it more obvious that sending is fallible.

Reply:
That sounds better to me, and is more in-line with what the stdlib receiver API does.

I'm also not sure that `sync_command` is the best name, since to me it sounds like an API that will wait for the command to complete on the same thread. `blocking_command` isn't a great name either for the same reason.

Maybe we should call this `spawn_command` or something like that?

Reply:
Senders now return a `Result` just like every other channel implementation. Where possible, methods like `input()` or `output()` unwrap the result automatically to greatly reduce the need for unwraps by the user. 

Also, `SharedState` and `Reducer` are now aware that sending may fail if a subscribed component is shut down.